llvm/unittests/Transforms/IPO/SampleProfileMatcherTests.cpp

```diff
@@ -0,0 +1,150 @@
+//===- SampleProfileMatcherTests.cpp - SampleProfileMatcher Unit Tests -----==//
```

Comment:
Thinking more about it, I feel we don't really need this level of unit test? There are a lot algorithms in the compiler, and when developing these algorithms it's important to test the core of it, but I don't know if we need to have this level of unit test committed in tree. Think about profi, or even layout algorithm, or many more, we usually test e2e, rather than creating tests for internals of the algorithms. 

I hope we can turn some of this into lit tests and also remove the debug / test functions in diffing.

Reply:
Got it, removed this unit test and the debug things.